### PR TITLE
Gcloud lib integration

### PR DESCRIPTION
**Pull request by Ressetkk**
**Description**

Changes proposed in this pull request:

- update `kyma-gke-integration.sh` to utilise `gcloud::` functions
-  fix typo in `gcloud::provision_gke_cluster`
- deprecate some of `prow/scripts/cluster-integration/helpers` in favour of `gcloud::` alternatives:
    - `encrypt` and `decrypt`
    - `create-dns-record` and `delete-dns-record`
    - `reserve-ip-address` and `release-ip-address`
    - `create-network-with-subnet` and `network-exists`
    - `provision-gke-cluster` and `deprovision-gke-cluster`
- deprecate some of `prow/scripts/cluster-integration/helpers` in favour of `utils::` alternatives:
    - `generate-self-signed-cert` with `utils::generate_self_signed_cert`

**Related issue(s)**

#2974

### CODE REVIEW

File: prow/scripts/cluster-integration/compass-gke-integration.sh [hunk 1]
```diff
@@ -158,7 +153,8 @@ function createCluster() {
     export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
   fi
   CLEANUP_CLUSTER="true"
-  env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+  export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+  gcloud::provision_gke_cluster "$CLUSTER_NAME"
```
Comment (tehcyx) [inline prow/scripts/cluster-integration/compass-gke-integration.sh:157]: This seems to hide functionality, couldn't we have this more explizit by doing something like this instead?
```sh
gcloud::provision_gke_cluster "${CLUSTER_NAME}" "${ADDITIONAL_LABELS}"
```

File: prow/scripts/cluster-integration/control-plane-gke-integration.sh [hunk 2]
```diff
@@ -171,7 +166,8 @@ function createCluster() {
     export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
   fi
   CLEANUP_CLUSTER="true"
-  env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+  export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+  gcloud::provision_gke_cluster "$CLUSTER_NAME"
```
Comment (tehcyx) [inline prow/scripts/cluster-integration/control-plane-gke-integration.sh:170]: See comment above for the the similar line.

File: prow/scripts/cluster-integration/helpers/create-cluster.sh [hunk 3]
```diff
@@ -62,7 +65,8 @@ function createCluster() {
 	if [ -z "${CLUSTER_VERSION}" ]; then
 		export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
 	fi
-	env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+	export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+	gcloud::provision_gke_cluster "$CLUSTER_NAME"
```
Comment (tehcyx) [inline prow/scripts/cluster-integration/helpers/create-cluster.sh:69]: see above

File: prow/scripts/cluster-integration/kyma-gke-compass-integration.sh [hunk 4]
```diff
@@ -178,7 +173,8 @@ function createCluster() {
     export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
   fi
   CLEANUP_CLUSTER="true"
-  env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+  export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+  gcloud::provision_gke_cluster "$CLUSTER_NAME"
```
Comment (tehcyx) [inline prow/scripts/cluster-integration/kyma-gke-compass-integration.sh:177]: see above

File: prow/scripts/cluster-integration/kyma-gke-long-lasting.sh [hunk 5]
```diff
@@ -90,28 +90,24 @@ source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/docker.sh"
...
 
 	log::info "Provision cluster: \"${CLUSTER_NAME}\""
 	date
 	
 	if [ -z "${CLUSTER_VERSION}" ]; then
 		export CLUSTER_VERSION="${DEFAULT_CLUSTER_VERSION}"
 	fi
-	env ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}" "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}"/provision-gke-cluster.sh
+	export ADDITIONAL_LABELS="created-at=${CURRENT_TIMESTAMP}"
+	gcloud::provision_gke_cluster "$CLUSTER_NAME"
```
Comment (tehcyx) [inline prow/scripts/cluster-integration/kyma-gke-long-lasting.sh:110]: see above

**Review by tehcyx — COMMENTED** [review]
I see the networking changes removed the message `log::info "Network ${GCLOUD_NETWORK_NAME} exists"` is this something that we still get from somewhere else? I believe this message is really helpful in case a network does indeed exist already for us to manually intervene, what do you think?

**Review by tehcyx — APPROVED** [review]

**Comment by Halamix2** [comment]
Chech and log::info that network already exists was integrated into `gcloud::create_network`

File: prow/scripts/cluster-integration/compass-gke-integration.sh (hunk 1 above)
Comment (Halamix2) [reply prow/scripts/cluster-integration/compass-gke-integration.sh:157]: There is problem with optional arguments: adding something like optional `$1` or `$@` in a function body and not adding it to any of the function call (e.g. just calling it by `cloud::provision_gke_cluster "${CLUSTER_NAME}"`) will make shellcheck complain and fail. Probably only way do do it it to disable one of the shellchecks for the whole function:
https://github.com/koalaman/shellcheck/wiki/SC2120

Comment (tehcyx) [reply prow/scripts/cluster-integration/compass-gke-integration.sh:157]: I'm thinking more of a change like this:
```sh
function gcloud::provision_gke_cluster {
  if [ -z "$1" ]; then
    log::error "Cluster name not provided. Provide proper cluster name."
    exit 1
  fi
  CLUSTER_NAME=$1
  if [ -z "$2" ]; then
    log::info "No additional labels provided"
  fi
  ADDITIONAL_LABELS=$2
```

This way you can still use it like this: `cloud::provision_gke_cluster "${CLUSTER_NAME}"` and from my a quick test shellcheck would not complain here, unless I'm missing something?

open https://www.shellcheck.net/, use this code:
```sh
#!/usr/bin/env bash

function gcloud::provision_gke_cluster {
  if [ -z "$1" ]; then
    log::error "Cluster name not provided. Provide proper cluster name."
    exit 1
  fi
  CLUSTER_NAME=$1
  if [ -z "$2" ]; then
    log::info "No additional labels provided"
  fi
  ADDITIONAL_LABELS=$2
  echo "${CLUSTER_NAME}"
  echo "${ADDITIONAL_LABELS}"
}

gcloud::provision_gke_cluster "hello"

gcloud::provision_gke_cluster "hello" "world"
```

Also the comment on the function does not indicate that providing `ADDITIONAL_LABELS` will have influence on anything that this function creates.

**Pull request merged**